[Yeh Yu Chun] iP

peter-yeh opened the pull request.

Commit:
Add Gradle support

peter-yeh committed:
Greet, Echo, Exit

peter-yeh committed:
Add, list

peter-yeh committed:
Mark as Done: added task class and mark items as done

peter-yeh committed:
streamlined the tasks type by introducing todo, events, deadlines

peter-yeh committed:
changed runtest.bat and expected.txt to the correct output

peter-yeh committed:
Handle Errors

peter-yeh committed:
Added support for delete keyword

peter-yeh committed:
changed the format of the string output

peter-yeh committed:
updated the expected.txt and input.txt to test delete

peter-yeh committed:
Update switch statement indentation level

Removed indentation for case clauses to match the style guide.

peter-yeh committed:
Add Javadocs comment

Added comments for all public methods to comply with java coding
standard

peter-yeh committed:
Add read from storage

Add DukeFileHandler which would read tasks stored in the storage and
display it

peter-yeh committed:
Update gitignore to ignore data file

The data/dukeData.txt is added to gitignore as it does not need to be
pushed

peter-yeh committed:
Add write list to file

Tasks would now be saved to the storage.

Tasks is now able to persist through sessions, remembering what the user
added to the task

peter-yeh committed:
Ignore files in data folder

The local data in the folder would not be uploaded to git

peter-yeh committed:
Add fix date format

Users would now have to type in a fixed date format. This would allow
the use of date for sorting tasks later on.

peter-yeh committed:
Merge branch 'branch-Level-7' into master

peter-yeh committed:
Merge branch 'branch-Level-8' into master

peter-yeh committed:
Update introducton message

Introduction message now inludes valid commands to be given to duke,
guiding users to type message with the correct format

peter-yeh committed:
Update welcome text

The welcome text is now more user friendly as the valid commands are
clearly listed to help users to give the correct format.

peter-yeh committed:
Add Ui class

This will handle all the displaying of inputs to the console.
It also includes private class to nicely format the output.

peter-yeh committed:
Remove most statc methods

Convert most of the static classes into a normal class. Duke would not
call the instatnce methods instead of the static methods.

Also added variable ui to be utilised later on

peter-yeh committed:
Update testing of Duke

Change the input.txt to the correct date format

Chnage the expected.txt to reflect the updated text format

peter-yeh committed:
Move welcome message into Ui class

Outputting to console will now solely be done in Ui class
This would reduce the workload of the Duke.java

peter-yeh committed:
Make storage more oop

Add constructor in DukeFileHandler class to take in the file path as a
string input

peter-yeh committed:
Create TaskList class

The task list is stored and handled inside the Duke class.

The Duke class is very cluttered with many methods to handle the tasks
list.

TaskList is created to handle the task related methods.

peter-yeh committed:
Add more testcase

The current test case does not test writing and reading from storage.

More testcase is written to test read and write function of Duke

peter-yeh committed:
Refactor the code to look more OOP

Created new object class called command with execute()  to handle
the different messages

peter-yeh committed:
Organize all the java classes into their package

The classes are separated into their packages to organize the codes

peter-yeh committed:
Add auto write to storage after every step

Before this, the user would have to enter bye before the list in the
storage would be updated.

The lists would now be saved in real time in case the user closes duke
without saying bye

peter-yeh committed:
Add Junit 5 to Task class

Testing is done for Task and it's subclasses

peter-yeh committed:
Change class name events to event

The name of the subclass events is changed to event

peter-yeh committed:
Add space to some methods

Spaces is added to comply with java coding standard

peter-yeh committed:
Add create dir if it does not exist

Write to file would throw error if the dir does not exist.

The program would now create a new dir and file if it does not exist

peter-yeh committed:
Add JavaDoc comments

All public classes except Command classes have javaDocs

peter-yeh committed:
Change codes to comply with java coding standard

peter-yeh committed:
Add findcommand class

User would be able to find tasks by using the find keyword

peter-yeh committed:
Include the add command option in the welcome message

Instuctions is added to teach the user how to use add command

The runtest.bat is not working

peter-yeh committed:
Merge tag 'A-JavaDoc' into master

Add JavaDocs

peter-yeh committed:
Merge branch 'branch-A-CodingStandard' into master

peter-yeh committed:
Merge branch 'branch-Level-9' into master

peter-yeh committed:
Merge branch 'add-gradle-support' into branch-A-Gradle

peter-yeh committed:
Merge remote-tracking branch 'origin/branch-A-Gradle' into master

peter-yeh committed:
Add checkstyle

Adjusted code to comply with checkstyle guide.

peter-yeh committed:
Add javaFx

The window is working

peter-yeh committed:
Can create new file

Txt file is now created if the folder is empty

peter-yeh committed:
Modify Duke class

Duke is now initialised later to allow the windows to initliase first.

Duke welcome message is now displayed after the windows is initialised.

peter-yeh committed:
Add feature to display output to window from ui class

System.out.println methods used in Ui class is now replaced by
window.showsToDialog method to be displayed to window.

peter-yeh committed:
Change to comply with java convention

Checkstyle is ran to check for java convention

peter-yeh committed:
Use getter to get window

Variale window in Main is changed to private to comply with java
convention.

Getter is used in Ui class to get window from Main class.

schoolex commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,70 @@
+
+    private Duke(String path) {
+        ui = new Ui();
+        ui.welcome();
```
Should the method name be revised to include a verb? i.e printWelcomeMsg()

schoolex commented on src/main/java/duke/utils/TaskList.java:
```diff
@@ -0,0 +1,112 @@
+     *
+     * @return the number of tasks in the list.
+     */
+    public int size() {
```
Should the method name be revised to include a verb? i.e getSize()

Avalionnet commented on src/main/java/duke/command/Command.java:
```diff
@@ -0,0 +1,14 @@
+ * Corresponds to commands the users wants to execute.
+ */
+public abstract class Command {
+    public abstract void execute(TaskList tasks, Ui ui, DukeFileHandler fileHandler) throws DukeException;
```
It seems you may have missed out on adding JavaDoc comments for the following method. Do remember to add them in when fine-tuning your file! 👍🏻

schoolex commented on src/main/java/duke/utils/Ui.java:
```diff
@@ -0,0 +1,89 @@
+     *
+     * @param s string to output to console
+     */
+    public void displayThis(String s) {
```
Should the param for this method be more explicit? i.e String msg

Avalionnet commented on src/main/java/duke/command/DeleteCommand.java:
```diff
@@ -0,0 +1,28 @@
+
+
+public class DeleteCommand extends Command {
+    private final int entryDelete;
```
Perhaps we can explore adopting a more expressive variable name here such as IndexToDelete to explain the purpose of this variable at a higher level of detail?

schoolex commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,70 @@
+import duke.utils.Parser;
+import duke.utils.TaskList;
+import duke.utils.Ui;
+
+
+
+public class Duke {
```
Should there be a Javadoc describing the class?

Avalionnet commented on src/main/java/duke/command/DoneCommand.java:
```diff
@@ -0,0 +1,27 @@
+
+
+public class DoneCommand extends Command {
+    private final int entryDone;
```
Similar to my previous comment in the DeleteCommand class, perhaps we can explore adopting a more specific variable name here as well! 😺

Avalionnet commented on src/main/java/duke/command/DoneCommand.java:
```diff
@@ -0,0 +1,27 @@
+    @Override
+    public void execute(TaskList tasks, Ui ui, DukeFileHandler fileHandler) throws DukeException {
+        try {
+            ui.displayThis("Nice! I've marked this task as done: \n        " + tasks.done(entryDone));
```
While non-essential, maybe you can consider adding this display string as a specific method in your UI class instead?

Adopting this strategy will help to shorten the code in your command classes as a simple call to maybe ui.showDoneMessage(entryDone); can be used instead of having the entire message displayed here. 

It will also be easier to modify your display messages if they all reside in the same UI class. Just a point to consider 😸  

Otherwise, it works fine as it is as well 👍🏻

Avalionnet commented on src/main/java/duke/task/Deadline.java:
```diff
@@ -0,0 +1,49 @@
+    }
+
+
+    public Deadline(String description, boolean isDone, String by) {
```
It seems you missed out on this constructor when writing the JavaDoc for this class. Do remember to add it in when you're fine-tuning your code!

Avalionnet commented on src/main/java/duke/task/Event.java:
```diff
@@ -0,0 +1,49 @@
+     * Creates a type of task with timing
+     *
+     * @param description detail of the task
+     * @param at          time in yyyy-mm-dd
```
It seems there may be an extra tab in the description for this JavaDoc line.

peter-yeh replied:
> It seems there may be an extra tab in the description for this JavaDoc line.

oh I want  to align with the text for description variable so it's easier to see. Is this against the coding convention?

Avalionnet commented on src/main/java/duke/task/Task.java:
```diff
@@ -0,0 +1,68 @@
+    }
+
+
+    protected String getStatusIcon() {
```
Perhaps it may be better to use the /** */ format for writing JavaDoc for this method instead of having it within the method itself?

Avalionnet commented on src/main/java/duke/utils/Ui.java:
```diff
@@ -0,0 +1,89 @@
+public class Ui {
+
+
+    public Ui() {
```
It seems you may have missed out on writing JavaDoc for this default constructor!

Avalionnet reviewed (commented):
It is evident that a lot of effort has been put into this project by the author! 
Excellent OOP design is apparent as well. Great job and keep it up 😸

schoolex commented on src/main/java/duke/utils/TaskList.java:
```diff
@@ -0,0 +1,112 @@
+     * @return List of tasks with the keyword.
+     */
+    public List<Task> findKeyword(String keyword) throws DukeException {
+        List<Task> containsKeywords = new ArrayList<>();
```
Should this be explicitly named tasksContainingKeywords instead?

schoolex reviewed (commented):
Well formatted but there are missing Javadocs. Some minor nits and it will be good to merge.

peter-yeh committed:
Update font

The font is updated to show duke logo properly.

peter-yeh committed:
Allow more text to be shown

The text is no longer truncated if it overflows

peter-yeh committed:
Move Test into src folder

peter-yeh committed:
Change user image

peter-yeh committed:
Improve formatting in fxml files

peter-yeh committed:
Add close window method

Typing bye would now close the window

peter-yeh committed:
Remove all commented codes

All System.out.println() is now replaced to output to the window.

The commented codes are all removed.

peter-yeh committed:
Merge branch 'branch-Level-10' into master

peter-yeh committed:
Change the code according to the review

Added changes as suggested by the pull request review

peter-yeh committed:
Use varargs to replace array

Some methods which takes in arrays are replaced by varargs.

peter-yeh committed:
Merge branch 'branch-A-Varargs' into master

peter-yeh committed:
Update JavaDocs

The javadocs were updated for some methods to reflect it's updated
behavior.

peter-yeh committed:
Change the image size

The UI is updated to make it look nicer

peter-yeh committed:
Change build.gradle to run the launcher class

Replace the run class in build.gradle

peter-yeh committed:
Move the test class into java folder

Set a java test source root folder to enable it to run with gradle

peter-yeh committed:
Move checkstyle into config folder

The checkstyle was not running properly because it is in the wrong
folder. Moved it to the config folder now.

Test classes were also changed to follow the java coding standard.

peter-yeh committed:
Turn on assertion in gradle

Add a line to enable assertion in build.gradle.

peter-yeh committed:
Move the methods in the Duke class.

peter-yeh committed:
Add assertion in DukeFileHandler

Assertions were used to ensure the tasks list is not empty.

peter-yeh committed:
Add assert statements to TaskList class

Assetions were used to ensure the methods were not called on empty
lists.

peter-yeh committed:
Add assertions to Ui class

Assertions were used to ensure the variables are in place before the
methods are run.

peter-yeh committed:
Use SLAP to abstract Tasks class

Abstracted methods which were commonly used and moved it to the Task
class, to reduce duplicity.

peter-yeh committed:
Adjust codes to comply with coding style

Added javaDocs to for methods which does not hava javaDocs.

peter-yeh committed:
Convert a for loop to stream

peter-yeh committed:
Merge pull request #1 from peter-yeh/branch-A-Assertions

Branch a assertions

peter-yeh committed:
Merge pull request #2 from peter-yeh/branch-A-CodeQuality

Branch a code quality

peter-yeh committed:
Merge pull request #3 from peter-yeh/branch-A-Streams

Convert a for loop to stream

peter-yeh committed:
Set up Continuous Integration(CI)

Add yml file for CI.

peter-yeh committed:
Merge branch 'branch-A-CI' into master

peter-yeh committed:
Add test case for parse

Implement testing for edge case where the input is null or space.

peter-yeh committed:
Add a list of aliases inside each command

The commands contains a list of acceptable strings to call that
command.

peter-yeh committed:
Add more alias for the commands

peter-yeh committed:
Add methods to check if keyword entered contains alias

Add enum class for commands, to use switch statement.

peter-yeh committed:
Add a new command type

Add unknown command type which would be returned if the keywords matched
none of the command type.

peter-yeh committed:
Update parse to switch through command Type

peter-yeh committed:
Add method to return the correct command type

Checks the keyword and return the command type  which matches the keyword.

peter-yeh committed:
Add support for a shortcut command for duke

peter-yeh committed:
Fix bug where the shortcut was wrongly called

D was used for both delete deadline.

peter-yeh committed:
Merge pull request #4 from peter-yeh/branch-C-FriendlierSyntax

Branch c friendlier syntax

peter-yeh committed:
Update code formatting

peter-yeh committed:
Match coding standard

peter-yeh committed:
Update greeting message.

The greeting message now contains shortcut keys for each command.

peter-yeh committed:
Fix bug for find and todo

The find shortcut is not working as the find description was specified
at substring index 5, it was changed to allow the find / f to be used.

The same issue also occured for todo, where it's shortcut, t, is not
working for string content less than 5.

peter-yeh committed:
Delete test case for Parser

The ParseTest was removed because it could not be properly tested.

peter-yeh committed:
Add catch to find and todo command

This will catch any error from happening to catch error from the user
input.

peter-yeh committed:
Move font style to fxml file

peter-yeh committed:
Widen the window to allow more text

peter-yeh committed:
Update the greeting message

peter-yeh committed:
Add font and make profile image circular

peter-yeh committed:
Change the image to square so it fits the circular theme

peter-yeh committed:
Fix bug for delete

Add catch statement to catch improper user input for delete command.

peter-yeh committed:
Add background colour

peter-yeh committed:
Reduce padding between text box

peter-yeh committed:
Add Screenshot of duke

peter-yeh committed:
Update the user guide

peter-yeh committed:
Remove text splitter class

The class is obsolete.

peter-yeh committed:
Update user guide

peter-yeh committed:
Add Images for user guide

peter-yeh committed:
Set theme jekyll-theme-minimal

peter-yeh committed:
Update README.md
